compiler/qsc/src/codegen/tests.rs

```diff
@@ -525,6 +525,145 @@ mod adaptive_profile {
         "#]].assert_eq(&qir);
     }
 
```

Comment:
Since this is a feature only meant for Adaptive profile, it would be worth adding a negative test where it is asserted that QIR generation fails when a custom measurement is included and you are targeting Base profile.